### PR TITLE
kubeadm: add experimental upgrade job

### PR DESCRIPTION
**Pull request by neolit123**
/sig cluster-lifecycle
/kind feature
/priority important-longterm
/assign @fabriziopandini 
/assign @spiffxp @cjwagner

### CODE REVIEW

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-upgrade.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    preset-dind-enabled: "true"
+    preset-kind-volume-mounts: "true"
+  decoration_config:
+    timeout: 2400000000000 #40m
+  extra_refs:
+  - org: kubernetes
+    repo: kubernetes
+    base_ref: master
+    path_alias: k8s.io/kubernetes
+  path_alias: k8s.io/kubeadm
```
Comment (neolit123) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-upgrade.yaml]: i have a quick question here.
if i remove this line where would my `$(pwd)` be?
i.e. how would that affect:
```
      - "./kinder/ci/kinder-run.sh"
```
i tried finding the documentation for that, but i couldn't .

**Comment by fabriziopandini** [comment]
Thanks @neolit123, looking forward to seeing first test results
/lgtm

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-upgrade.yaml (hunk 1 above)
Comment (spiffxp) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-upgrade.yaml]: The docs are here, I can't say I actually grok them enough to answer your question though: https://github.com/kubernetes/test-infra/blob/master/prow/pod-utilities.md

Relevant lines seem to be 

>`periodic` jobs will have the working directory set to the root of the repo specified by the first ref in `extra_refs` (if specified)

>Repos requiring a non-standard clone path can use the `path_alias` field
to clone the repo to different go import path than the default of `/home/prow/go/src/github.com/{{.Org}}/{{.Repo}}/` (e.g. `path_alias: k8s.io/test-infra` -> `/home/prow/go/src/k8s.io/test-infra`)

Comment (neolit123) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-upgrade.yaml]: thanks, i read that page but missed the part about the first extra_ref!

**Pull request merged**